### DISCUSSION TITLE
Problems to build firmware 7.7

### DISCUSSION THREAD

**Discussion by pableitordasauto** [shanteacontrols/OpenDeck#276] [General]
Hi,
I'm trying to build the firmware 7.7
I'm not expert in Ubuntu. I have followed the instructions.
1- Clone the git repository
2- Create the docker container. Then, i get the following error:"Error responde from daemon: Head "https://ghcr.io/v2/shanteacontrols/opendeck/manifests/latest": unauthorized"

And I can't go forward.
Any idea?
Thanks

**Comment by paradajz** [comment]
It probably means your user account isn't in `docker` group. Check this guide:
https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-22-04#step-2-executing-the-docker-command-without-sudo-optional

Then again it's not clear to me if the error you're getting is related to that or permission to pull the Docker image (which should be public).

**Comment by paradajz** [comment]
No, sorry, the problem was the image visibility. You should be able to pull it now.

**Comment by pableitordasauto** [comment]
Now, the docker is working but I have error when I make a target. I don`t remember this problem in the previous version.
This is the output:

pablo@pablo-NITX-SKL1:~/OpenDeck$ make TARGET=mega2560
Generating CMake files
make[1]: *** /modules/core: No existe el archivo o el directorio.  Alto.
make[1]: *** /modules/core: No existe el archivo o el directorio.  Alto.
make[1]: *** /modules/core: No existe el archivo o el directorio.  Alto.
make[1]: *** /modules/core: No existe el archivo o el directorio.  Alto.
CMake Error at src/firmware/board/CMakeLists.txt:5 (include):
  include could not find requested file:

    /home/pablo/OpenDeck/src/generated/target/mega2560/CMakeLists.txt
Call Stack (most recent call first):
  CMakeLists.txt:37 (include)


CMake Error at src/firmware/board/CMakeLists.txt:19 (target_compile_definitions):
  Cannot specify compile definitions for target "mcu" which is not built by
  this project.
Call Stack (most recent call first):
  CMakeLists.txt:37 (include)


CMake Error at cmake/include/helpers.cmake:3 (add_subdirectory):
  add_subdirectory given source "/modules/LESSDB" which is not an existing
  directory.
Call Stack (most recent call first):
  src/firmware/application/CMakeLists.txt:41 (add_module)
  CMakeLists.txt:39 (include)


CMake Error at cmake/include/helpers.cmake:8 (target_include_directories):
  Cannot specify include directories for target "lessdb-lib" which is not
  built by this project.
Call Stack (most recent call first):
  src/firmware/application/CMakeLists.txt:42 (add_mcu_library)
  CMakeLists.txt:39 (include)


CMake Error at cmake/include/helpers.cmake:3 (add_subdirectory):
  add_subdirectory given source "/modules/SysExConf" which is not an existing
  directory.
Call Stack (most recent call first):
  src/firmware/application/CMakeLists.txt:44 (add_module)
  CMakeLists.txt:39 (include)


CMake Error at cmake/include/helpers.cmake:8 (target_include_directories):
  Cannot specify include directories for target "sysexconf-lib" which is not
  built by this project.
Call Stack (most recent call first):
  src/firmware/application/CMakeLists.txt:45 (add_mcu_library)
  CMakeLists.txt:39 (include)


CMake Error at cmake/include/helpers.cmake:3 (add_subdirectory):
  add_subdirectory given source "/modules/MIDI" which is not an existing
  directory.
Call Stack (most recent call first):
  src/firmware/application/CMakeLists.txt:47 (add_module)
  CMakeLists.txt:39 (include)


CMake Error at cmake/include/helpers.cmake:8 (target_include_directories):
  Cannot specify include directories for target "midi-lib" which is not built
  by this project.
Call Stack (most recent call first):
  src/firmware/application/CMakeLists.txt:48 (add_mcu_library)
  CMakeLists.txt:39 (include)


-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Configuring incomplete, errors occurred!
See also "/home/pablo/OpenDeck/build/mega2560/release/CMakeFiles/CMakeOutput.log".
make: *** [Makefile:31: cmake_config] Error 1

**Comment by paradajz** [comment]
It looks like you aren't building the firmware inside the container.

**Comment by pableitordasauto** [comment]
With previous versions I didn't have any problem...following the instructions it works always...I don't know whats happens.
My steps are:
1- Erase the Opendeck folder
2- Follow the install instructions
3- Build the target

I have a backup folder with the previous versions. Is there any other manner to update the firmware without the previous process described?

**Comment by paradajz** [comment]
Well now the procedure is different and the documentation makes it pretty clear. You need to run the mentioned script after which you'll be inside the container. You are not building the firmware in the container.

**Comment by pableitordasauto** [comment]
Ok. inside the container it works. Now i have a doubt...How I did before to build outside the container?
And (sorry for my poor linux knowledge) How I can copy the merged.uf2 file outside the container?
I'm very lost...

**Comment by paradajz** [comment]
It worked because the repository was setup differently than it is now. The merged file is still located in the build directory which you can still access the same way as before.

**Comment by pableitordasauto** [comment]
Hi again. I was some days missing...now I'm trying to build the firmware again.
I'm a a dissaster. I see a new ver. 7.7.1. 
I downloaded the repository and now, when I run ./scripts/dev.sh the message from the system is:

pablo@pablo-NITX-SKL1:~/OpenDeck$ ./scripts/dev.sh
Creating new container
[+] Pulling 1/1
 ✔ opendeck Pulled                                                                              0.9s 
[+] Running 0/0
 ⠋ Container opendeck  Starting                                                                 0.0s 
Error response from daemon: failed to create endpoint opendeck on network opendeck_default: network 2a4a641e444397f8f236471f6423e162a99173744fd0f5e6f04d3d1dde317d5d does not exist

I have update all docker modules, ubuntu...but I don't know where is the problem
